### PR TITLE
Null/undefined check editors

### PR DESCRIPTION
**Pull request by aaomidi**
BaseEditor specifies that it could potentially return a null editor:

https://github.com/microsoft/azuredatastudio/blob/master/src/vs/workbench/browser/parts/editor/baseEditor.ts#L58-L60

 Not sure if we should treat it the same, so I opened this PR as a discussion point.

I've noticed that a few of our issues are that we hit editor is not defined when we try to access the `uri` property of it.

### CODE REVIEW

**Comment by aaomidi** [comment]
I used console.error to get the printed stacktrace, might help us with debugging even more :)

**Comment by anthonydresser** [comment]
The editor can be null, but in those actions they shouldn't be null since those actions can only be ran on an active editor. So there must be an editor when those are running, so sounds like a bug.

File: src/sql/workbench/parts/query/browser/queryActions.ts [hunk 1]
```diff
@@ -552,6 +556,10 @@ export class ToggleSqlCmdModeAction extends QueryTaskbarAction {
 		// set query options
 		let queryoptions: QueryExecutionOptions = { options: new Map<string, any>() };
 		queryoptions.options['isSqlCmdMode'] = toSqlCmdState;
+		if (!this.editor.input) {
+			console.error('editor input was null');
+			return Promise.resolve(null);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryActions.ts]: Should just be Promise.resolve() - this method returns Promise<void>

File: src/sql/workbench/parts/query/browser/queryActions.ts [hunk 2]
```diff
@@ -711,6 +724,11 @@ export class ListDatabasesActionItem implements IActionViewItem {
 	}
 
 	private getCurrentDatabaseName() {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryActions.ts]: Could you update this to add the return type string | undefined while you're here?

File: src/sql/workbench/parts/query/browser/queryActions.ts [hunk 3]
```diff
@@ -734,6 +752,11 @@ export class ListDatabasesActionItem implements IActionViewItem {
 			return;
 		}
 
+		if (!this._editor.input) {
+			console.error('editor input was null');
+			return undefined;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryActions.ts]: Just return - this method doesn't have a return type.

File: src/sql/workbench/parts/query/browser/queryActions.ts [hunk 4]
```diff
@@ -743,6 +766,11 @@ export class ListDatabasesActionItem implements IActionViewItem {
 	}
 
 	private onDropdownFocus(): void {
+		if (!this._editor.input) {
+			console.error('editor input was null');
+			return undefined;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryActions.ts]: Just return - this method doesn't have a return type.

File: src/sql/workbench/parts/query/browser/queryActions.ts [hunk 5]
```diff
@@ -278,6 +278,10 @@ export class CancelQueryAction extends QueryTaskbarAction {
 
 	public run(): Promise<void> {
 		if (this.isConnected(this.editor)) {
+			if (!this.editor.input) {
+				console.error('editor input was null');
+				return Promise.resolve(null);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryActions.ts]: Should just be Promise.resolve() - this method returns Promise\<void>

Comment (aaomidi) [reply src/sql/workbench/parts/query/browser/queryActions.ts]: I'm following existing code here to make my changes as minimal as possible: https://github.com/microsoft/azuredatastudio/blob/f4c2e28262f5cf81c811ecb99fbc09718fc7703d/src/sql/workbench/parts/query/browser/queryActions.ts#L219

File: src/sql/workbench/parts/query/browser/queryActions.ts (hunk 1 above)
Comment (aaomidi) [reply src/sql/workbench/parts/query/browser/queryActions.ts]: Following the flow of the code again here:

https://github.com/microsoft/azuredatastudio/blob/f4c2e28262f5cf81c811ecb99fbc09718fc7703d/src/sql/workbench/parts/query/browser/queryActions.ts#L567

File: src/sql/workbench/parts/query/browser/queryActions.ts (hunk 5 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/parts/query/browser/queryActions.ts]: I'd actually suggest fixing that up too then while you're here. What it's doing right now is just wrong and at best is confusing and at worst could cause unintended behavior if someone sees that and expects to get something back.

File: src/sql/workbench/parts/query/browser/queryActions.ts [hunk 6]
```diff
@@ -471,15 +473,15 @@ export class ToggleConnectDatabaseAction extends QueryTaskbarAction {
 		}
 	}
 
-	public run(): Promise<void> {
+	public async un(): Promise<void> {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/query/browser/queryActions.ts]: >un [](start = 14, length = 2)

Deleted an r!

File: src/sql/workbench/parts/query/browser/queryActions.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/parts/query/browser/queryActions.ts]: bump - it'd still be good to add the return type here

---
In reply to: [322376193](https://github.com/microsoft/azuredatastudio/pull/7084#discussion_r322376193) [](ancestors = 322376193)

File: src/sql/workbench/parts/query/browser/queryActions.ts (hunk 3 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/parts/query/browser/queryActions.ts]: You're still returning undefined here - this should just be return;

---
In reply to: [322376405](https://github.com/microsoft/azuredatastudio/pull/7084#discussion_r322376405) [](ancestors = 322376405)

File: src/sql/workbench/parts/query/browser/queryActions.ts (hunk 6 above)
Comment (aaomidi) [reply src/sql/workbench/parts/query/browser/queryActions.ts]: Oh no how did I not notice this.

**Comment by aaomidi** [comment]
Sorry it took me some time to get these changes out, this issue got hidden under the other stuff I was working on.

**Pull request merged**